File: src/circle.js
// /**
//  * (c) 2016 Jörn Dinkla, www.dinkla.net
//  * See the file LICENSE in the ROOT directory.
//  *
//  * Ein Kreis bzw. ein "Planet". Siehe Abschnitt 2.3 im Buch.
//  */

import Vector from '../src/vector';

class Circle {

  constructor(color = '', mass = 0.0, position = Vector.origin, velocity = Vector.origin) {
    this.color = color;
    this.mass = mass;
    this.position = position;
    this.velocity = velocity;
  }

  step() {
    this.position = this.position.plus(this.velocity);
  }

}

export default Circle;

File: lib/circle.dart
/**
 * A circle or "planet". See Section 2.3 in the book.
 */

import 'package:complexity_perspective/vector.dart';

class Circle {

  String color;
  num mass;
  Vector position;
  Vector velocity;

  Circle({this.color = "", this.mass = 0.0, this.position = Vector.origin, this.velocity = Vector.origin});

  step() {
    position = position + velocity;
  }

  String toString() => "color: $color, mass: $mass, position: $position, velocity: $velocity";

}